flink-state-backends/flink-statebackend-changelog/src/test/java/org/apache/flink/state/changelog/ChangelogDelegateHashMapTest.java

```diff
@@ -80,4 +80,12 @@ public void testMaterializedRestore() throws Exception {
         ChangelogStateBackendTestUtils.testMaterializedRestore(
                 getStateBackend(), env, streamFactory);
     }
+
+    @Test
+    public void testMaterializedRestorePriorityQueue() throws Exception {
```

Comment:
Do we really need to add this method to every descendant of `HashMapStateBackendTest`? Can't we only add it to `HashMapStateBackendTest`?

Reply:
Okay I see there's a dependency problem.

Reply:
There are two reasons (we discussed when testMaterializedRestore)

1. dependency
2. should expose a way to ignore trigger materialization because non-changelog does not support trigger materialization